tracing-opentelemetry/src/main/java/io/micronaut/tracing/opentelemetry/http/OpenTelemetryServerFilter.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.tracing.opentelemetry.http;
+
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.http.HttpHeaders;
+import io.micronaut.http.HttpRequest;
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.MutableHttpHeaders;
+import io.micronaut.http.MutableHttpResponse;
+import io.micronaut.http.annotation.Filter;
+import io.micronaut.http.filter.HttpServerFilter;
+import io.micronaut.http.filter.ServerFilterChain;
+import io.micronaut.tracing.opentelemetry.util.TracingObserver;
+import io.micronaut.tracing.opentelemetry.util.TracingPublisherUtils;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanBuilder;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.propagation.TextMapGetter;
+import io.opentelemetry.context.propagation.TextMapSetter;
+import jakarta.inject.Inject;
+import org.reactivestreams.Publisher;
+
+import static io.micronaut.http.filter.ServerFilterPhase.TRACING;
+import static io.micronaut.tracing.opentelemetry.http.AbstractOpenTracingFilter.SERVER_PATH;
+import static io.micronaut.tracing.opentelemetry.http.TraceRequestAttributes.CURRENT_SPAN;
+import static io.micronaut.tracing.opentelemetry.http.TraceRequestAttributes.CURRENT_SPAN_CONTEXT;
+
+/**
+ * An HTTP server instrumentation filter that uses Open Telemetry.
+ *
+ * @author Nemanja Mikic
+ * @since 1.0
+ */
+@Filter(SERVER_PATH)
+@Requires(beans = Tracer.class)
+public class OpenTelemetryServerFilter extends AbstractOpenTracingFilter implements HttpServerFilter {
```

Comment:
Please take a look at this example which reuse official `opentelemetry-instrumentation-api` library: https://github.com/tailrocks/micronaut-opentelemetry/blob/7f1e83e70e023375b3fe6684a8c273fc649d1682/opentelemetry/src/main/java/io/micronaut/opentelemetry/instrumentation/http/server/OpenTelemetryServerFilter.java#L42